planner/core/find_best_task.go

```diff
@@ -936,26 +938,90 @@ func (ds *DataSource) buildIndexMergeTableScan(prop *property.PhysicalProperty,
 	return ts, partialCost, nil
 }
 
-func indexCoveringCol(col *expression.Column, indexCols []*expression.Column, idxColLens []int) bool {
+func indexCoveringCol(col *expression.Column, constVal *expression.Constant, indexCols []*expression.Column, idxColLens []int) bool {
```

Comment:
maybe we could let it be called by `extractColumnsAndConstants`. So we can avoid creating new slices.